### PR TITLE
feat(id2iri): replace IDs also inside salsah-links, not only inside <resptr> tags (DEV-2578)

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

File: src/dsp_tools/utils/id_to_iri.py [hunk 1]
```diff
@@ -73,16 +76,35 @@ def _replace_ids_by_iris(
...
+        value_before = resptr_elem.text
+        if value_after := mapping.get(value_before):
+            resptr_elem.text = value_after
+            resptr_elems_replaced += 1
+            used_mapping_entries.add(value_before)
+
+    salsah_links = [
+        x for x in tree.xpath("/knora/resource/text-prop/text//a") if x.attrib.get("class") == "salsah-link"
+    ]
+    salsah_links_replaced = 0
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/utils/id_to_iri.py:93]: I would make salsah and resptr_elems two separate functions.

File: test/e2e/test_cli.py [hunk 2]
```diff
@@ -205,15 +205,17 @@ def test_xml_upload_incremental(self) -> None:
...
-        mapping_file = list(Path().glob("test-data-systematic_id2iri_mapping_*.json"))[0]
+        mapping_file_candidates = Path().glob("test-data-systematic_id2iri_mapping_*.json")
+        mapping_file = sorted(mapping_file_candidates)[-1]  # most recent one
         second_xml_file_orig = Path("testdata/id2iri/test-id2iri-data.xml")
         self._make_cli_call(f"dsp-tools id2iri {second_xml_file_orig.absolute()} {mapping_file.absolute()}")
+        mapping_file.unlink()
 
-        second_xml_file_replaced = list(self.cwd.glob(f"{second_xml_file_orig.stem}_replaced_*.xml"))[0]
+        second_xml_file_replaced_candidates = self.cwd.glob(f"{second_xml_file_orig.stem}_replaced_*.xml")
+        second_xml_file_replaced = sorted(second_xml_file_replaced_candidates)[-1]  # most recent one
```
Comment (Nora-Olivia-Ammann) [inline test/e2e/test_cli.py:215]: You are doing the same thing twice 208-212 and 214 and again in 268, maybe you could make a function for this?

File: test/unittests/test_id_to_iri.py [hunk 3]
```diff
@@ -39,24 +39,29 @@ def test_invalid_json_file_name(self) -> None:
             )
 
     def test_replace_id_with_iri(self) -> None:
+        """Check that the correct IRIs appear in the correct order in the output file"""
         id_to_iri(
             xml_file="testdata/id2iri/test-id2iri-data.xml",
             json_file="testdata/id2iri/test-id2iri-mapping.json",
         )
-        out_file = list(Path(".").glob("test-id2iri-data_replaced_*.xml"))[0]
```
Comment (Nora-Olivia-Ammann) [inline test/unittests/test_id_to_iri.py:46]: Same with the path as in file test_cli.py

**Review by Nora-Olivia-Ammann — CHANGES_REQUESTED** [review]

File: src/dsp_tools/utils/id_to_iri.py (hunk 1 above)
Comment (jnussbaum) [reply src/dsp_tools/utils/id_to_iri.py:112]: better now?

File: test/e2e/test_cli.py (hunk 2 above)
Comment (jnussbaum) [reply test/e2e/test_cli.py]: better now?

**Review by Nora-Olivia-Ammann — APPROVED** [review]

**Pull request merged**